bioblend/galaxy/histories/__init__.py

```diff
@@ -662,27 +662,28 @@ def download_history(self, history_id, jeha_id, outf,
         for chunk in r.iter_content(chunk_size):
             outf.write(chunk)
 
-    def copy_dataset(self, history_id, dataset_id, source='hda'):
+    def copy_dataset(self, history_id, history_content_id, source='hda'):
```

Comment:
Instead of modifying the `copy_dataset()` method, I'd reimplement it as a simple call to a new, more general, `copy_content()` method:

```suggestion
    def copy_dataset(self, history_id, dataset_id, source='hda'):
        """
        Copy a dataset to a history.

        :type history_id: str
        :param history_id: history ID to which the dataset should be copied

        :type dataset_id: str
        :param dataset_id: dataset ID

        :type source: str
        :param source: Source of the dataset to be copied: 'hda' (the default), 'library' or 'library_folder'

        :rtype: dict
        :return: Information about the copied dataset
        """
        return self.copy_content(history_id, dataset_id, source)


    def copy_content(self, history_id, content_id, source='hda'):
```